RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotFromDB.cc

```diff
@@ -0,0 +1,80 @@
+/**_________________________________________________________________
+   class:   OnlineBeamSpotFromDB.cc
+   package: RecoVertex/BeamSpotProducer
+   
+
+
+ author: Francisco Yumiceva, Fermilab (yumiceva@fnal.gov)
+
+
+________________________________________________________________**/
+
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/IOVSyncValue.h"
+#include "CondFormats/DataRecord/interface/BeamSpotTransientObjectsRcd.h"
+#include "CondFormats/BeamSpotObjects/interface/BeamSpotOnlineObjects.h"
+#include "CondFormats/BeamSpotObjects/interface/BeamSpotObjects.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include <string>
+class MyHelper {
+public:
+  MyHelper(edm::ParameterSet const& iPS,
+           edm::ConsumesCollector&& iC,
+           edm::ESGetToken<BeamSpotObjects, BeamSpotTransientObjectsRcd>& bsToken_) {
+    bsToken_ = iC.esConsumes<BeamSpotObjects, BeamSpotTransientObjectsRcd>();
+  }
+};
+
+class OnlineBeamSpotFromDB : public edm::one::EDAnalyzer<> {
+public:
+  explicit OnlineBeamSpotFromDB(const edm::ParameterSet& iConfig);
+  ~OnlineBeamSpotFromDB() override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& desc);
+  edm::ESGetToken<BeamSpotObjects, BeamSpotTransientObjectsRcd> bsToken_;
+
+private:
+  void beginJob() override;
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+  void endJob() override;
+};
+
+OnlineBeamSpotFromDB::OnlineBeamSpotFromDB(const edm::ParameterSet& iConfig) {
+  MyHelper m_helper(iConfig, consumesCollector(), bsToken_);
+}
```

Comment:
> Do you have an example for using the the esConsumes from an EDAnalyzer without the need of an helper function?

I'm confused. Could you point the documentation where you found the helper functions (that may need to be improved)?

Here all can be done without `MyHelper` by just
```suggestion
OnlineBeamSpotFromDB::OnlineBeamSpotFromDB(const edm::ParameterSet& iConfig)
  : bsToken_(esConsumes<BeamSpotObjects, BeamSpotTransientObjectsRcd>)
  {}
```